thunder/executors/nvfuserex_impl.py

```diff
@@ -325,7 +374,33 @@ def translate_bound_symbol(bsym: BoundSymbol) -> Any:
             nvout = lc_to_nv_map[out]
             fd.add_output(nvout)
 
-    return fd
+    bind(fd, definition, "definition")
+
+    is_multigpu_fd = False
+    if any(map(lambda t: isinstance(t, DTensorProxy), sorted_unique_inputs)):
+        # multi-GPU path
+        assert all(map(lambda t: isinstance(t, DTensorProxy), sorted_unique_inputs)), (
+            "nvfuser: Currently we only support Fusion region with all DTensor inputs or all Tensor inputs but not a mix"
+        )
+
+        def check_dtensor_tracing_and_runtime_metadata(inp):
+            x, y = inp
+            _, _, dtensor_metadata = y
+            runtime_device_mesh_repr = dtensor_metadata[0]
+            runtime_placements_repr = dtensor_metadata[1]
+            return repr(x.device_mesh) == runtime_device_mesh_repr and repr(x.placements) == runtime_placements_repr
+
+        assert all(map(check_dtensor_tracing_and_runtime_metadata, zip(sorted_unique_inputs, input_descriptors))), (
+            "nvfuser: Expected runtime and tracing metadata to be the same for DTensor."
+        )
+
+        _find_tensor_by_index, multidevice_schedule = get_methods_for_dtensor_fd(sorted_unique_inputs)
+
+        bind(fd, multidevice_schedule, "multidevice_schedule")
+        bind(fd, _find_tensor_by_index, "_find_tensor_by_index")
+        is_multigpu_fd = True
+
+    return fd, is_multigpu_fd
```

Comment:
```suggestion
    return fd
```
is_multigpu_fd is essentially hasattr(fd, "multidevice_schedule"). It doesn't need to be returned